### ISSUE TITLE
Add filters to Team Pledges page

### ISSUE THREAD

**Issue by iandunn** [WordPress/five-for-the-future#129]
The Team Pledges pages list all people who've selected that they contribute to that team in their profile. This could be people who've just started volunteering, or people who are sponsored by their company. This could be used by team reps to proactively contact people to work on projects, but currently it's a very long list, which is hard to parse.

Examples: [Community](https://make.wordpress.org/community/pledges/), [Design](https://make.wordpress.org/design/pledges/).

We could add filters and/or sorting to help team reps navigate. Ideas are:

- Different sort options: alphabetical (default); by number of hours contributed…
- Default to showing only people with "[Team] Contributor" badges, and have a second view for "all contributors"
- Maybe a view for people who don't have the contributor badge (so reps can correct that if needed)

Other ideas or feedback on these? cc @melchoyce for eventual design input

-----

See https://github.com/WordPress/five-for-the-future/issues/69#issuecomment-563369697 and subsequent comments for some discussion.

**Comment by ryelle** [comment]
@iandunn What's the decision that's needed here? Can you recap the relevant bits of #69?

**Comment by iandunn** [comment]
- [ ] Another thing that might be helpful is to just automatically hide anyone who doesn't have a  Slack account connected to their profile, since there's no way that a team rep could contact them to get them involved in a project. 

_Note to self: See `bp_wporg_get_slack_username()`, but probably add to existing SQL queries so that it's performant._

---

Update: There's some code for this in `remove_unreachable_users()`, but it needs to be enabled in `get_team_contributors()` once there's a UI.

**Comment by andreamiddleton** [comment]
I think sort options would be really smart here. I agree about Alphabetical, number of hours pledged, and then maybe by badge type, if possible. 

It would also be cool to add a contributor badge icon to names that have badges that relate to the team (for Community team it would include meetup & wordcamp organizer, plus deputy and speaker). That latter idea would need some design work.

**Comment by iandunn** [comment]
> [maybe sort] by badge type, if possible.

I don't see any barriers to doing that 👍 

I was thinking the UI would _filter_ them on/off, though, rather than _sorting_ them. That way the list would be much shorter and easier to work with.

Would you prefer sorting instead?

**Comment by iandunn** [comment]
Do you have any thoughts on [hiding people from the list until they add (Slack) contact info to their profile](https://github.com/WordPress/five-for-the-future/issues/129#issuecomment-567100219)?

**Comment by iandunn** [comment]
I played around with that idea, and hiding users that team reps can't contact would cut the list in half, so I think that'll be a pretty effective way to improve the UX.

**Comment by andreamiddleton** [comment]
> Do you have any thoughts on hiding people from the list until they add (Slack) contact info to their profile?

Short answer: I think it's a bad idea right now, but we could add a "sort by" variable for that. 

To learn more about my reasoning for that, here's my "stream of consciousness" on the topic, handily typed up. ;)

Certainly having a Slack account on wordpress.org isn't required to be a productive contributor, though of course it's necessary to participate in team meetings that happen in Slack (I'm assuming they all do, but I could be wrong there — it's possible there are some teams or groups that meet in IRC or via other Slack instances. 

I agree cutting the lists down will help the UX of someone looking for pledged contributors they might ping to recruit, but there isn't really anything standing in the way of someone clicking through to someone's profile and reaching out via other contact methods that the contributor has shared on their profile. But even if there's no way to reach out to someone on their wordpress.org profile, there isn't anything stopping someone from participating in Trac or on P2s and contributing that way, you know? Though I suppose we're not talking about pulling someone's pledge down, but just hiding  them on the team blog list...

it's possible that most teams would prefer to have people without a Slack account to be hidden on their list, but until we know that, I think we hold off and just add another sort variable. I'll ask this question on the post I plan on posting next week.

**Comment by iandunn** [comment]
> I think we hold off and just add another sort variable. I'll ask this question on the post I plan on posting next week.

👍 

I committed the code to filter/sort by it in `dotorg 15692`, but left it disabled until we have a UI for that.

**Comment by melchoyce** [comment]
We can use the same filter design used for 5ftF:

![image](https://user-images.githubusercontent.com/2846578/71834168-8db8d500-307c-11ea-8310-e9cc1e23b101.png)

**Comment by iandunn** [comment]
That looks like _sorting_ and _search_ to me, rather than _filtering_. Do you think that'd be better, or are we using different definitions?

**Comment by melchoyce** [comment]
I was assuming sorting. Do we think folks will need, or want, to filter by multiple different criteria?

**Comment by iandunn** [comment]
I would think so. If we only sort, then the list will still be ~600 items long, which can feel overwhelming, even if you know that the results you wanted start at the top of the list. 

You won't know where the  results you want end, and there will still be hundreds of results that match a single sorting factor.

With something like on/off toggles, you could check `has a slack account`, and the list would shrink from ~600 to ~300. Then, you could check `has contributed before`, and it'd shrink from ~300 to ~100, which is starting to get somewhat manageable (although it's still much longer than the ideal).

**Comment by melchoyce** [comment]
Sounds like we might be using filtering as a band-aid for the fact that there are so many false positives?

**Comment by iandunn** [comment]
I 💯agree, but it sounds we're going to [try this approach for now](https://github.com/WordPress/five-for-the-future/issues/129#issuecomment-569736752).

**Comment by melchoyce** [comment]
It's probably worth waiting to see what comments that post turns up (was it published, @andreamiddleton?) in case a better solution is presented.

One thing we could consider is calling out team members separately, then contributors with badges (since we have separate badges for team members vs. contributors), and then unbadged contributors in the last group.

**Comment by tellyworth** [comment]
Bumping this to add a more recent example that came up in p2 discussions:

https://make.wordpress.org/project/2022/03/16/proposal-to-remove-spam-dormant-five-for-the-future-pledges/#comment-54

**Proposed, with tabs to filter the list:**
![](https://make.wordpress.org/project/files/2022/03/Screen-Shot-2022-03-17-at-7.42.38-AM.png)

**Current, with only an unfiltered view sorted by name:**
<img width="705" alt="Screen Shot 2022-07-01 at 4 50 44 pm" src="https://user-images.githubusercontent.com/7200686/176840278-f79b2ec1-4717-464e-b92c-fff519a6bb31.png">

Any simple filters or tabs to help surface recently active pledges would be an improvement. Even just sorting the list by date and adding some pagination would help.
